webalyzer/collector/views.py

```diff
@@ -62,3 +71,25 @@ def collect(request):
         'OK', status=created and 201 or 200)
     response['Access-Control-Allow-Origin'] = '*'
     return response
+
+
+def collect_check(request, source_hash, source_type):
+    source_hash = int(source_hash)
+    matches = 0
+
+    # do we have a counter to increment?
+    # do we want to update the date?
+
+    if source_type == 'css':
+        matches = Stylesheet.objects.filter(source_hash=source_hash)
+    elif source_type == 'html':
+        matches = Page.objects.filter(source_hash=source_hash)
+
+    if len(matches) > 0:
```

Comment:
change to `if matches.exists():` which is faster as it might not need to do a count of how many when all you're interested in is if there 0 or not 0.